article/app/controllers/ArticleController.scala

```diff
@@ -117,16 +121,27 @@ class ArticleController extends Controller with RendersItemResponse with Logging
 
   def renderLiveBlog(path: String, page: Option[String] = None) =
     Action.async { implicit request =>
-      mapModel(path, blocks = true, page) {// temporarily only ask for blocks too for things we know are new live blogs until until the migration is done and we can always use blocks
-        render(path, _)
+      val range = page.map(ParseBlockId.fromPageParam) match {
+        case Some(Some(id)) => Left(PageWithBlock(id)) // we know the id of a block
+        case Some(None) => Right(NotFound) // page param there but couldn't extract a block id
+        case None => Left(Canonical) // no page param
+      }
+      range.left.map { range =>
+        mapModel(path, range = Some(range)) {// temporarily only ask for blocks too for things we know are new live blogs until until the migration is done and we can always use blocks
+          render(path, _)
+        }
+      } match {
+        case Left(f) => f
+        case Right(status) => Future.successful(Cached(10)(WithoutRevalidationResult(status)))
       }
     }
 
   def renderLiveBlogJson(path: String, lastUpdate: Option[String], rendered: Option[Boolean], isLivePage: Option[Boolean]) = {
     Action.async { implicit request =>
-      mapModel(path, blocks = true) { model =>
-        (lastUpdate, rendered) match {
-          case (Some(lastUpdate), _) => renderNewerUpdates(model, lastUpdate, isLivePage)
+      val range = lastUpdate.flatMap(blockId => ParseBlockId.fromBlockId(blockId)).orElse(rendered.flatMap(r => if (r == false) Some(Canonical) else None))
```

Comment:
actually it had a gap in the logic, I'm rewriting in line with your above suggestion